### ISSUE TITLE
Accessibility: Focus escapes modal when tabbing (focus trap not enforced)

### ISSUE THREAD

**Issue by MichalMG** [shopware/platform#15035]
### Shopware Version

6.7.x

### Affected area / extension

Platform(Default)

### Actual behaviour

For example, when opening the offcanvas cart (modal dialog), keyboard focus is not properly trapped within the modal. 
The focus trap appears to work only when navigating backwards using **Shift + Tab**, but fails when navigating forward using **Tab**. 
As a result, pressing **Tab** allows focus to move outside of the modal, which should not be possible while the modal/offcanvas is open.

Violates WCAG 2.4.3 (Focus Order)- relevant for WCAG 2.2, because keyboard focus inside the modal does not follow a predictable and logical sequence. This can cause confusion for keyboard-only users and may lead them to interact with elements outside the modal.

### Expected behaviour

Focus should remain trapped within the modal. 
When reaching the last focusable element and pressing **Tab**, focus should wrap to the first focusable element inside the modal.

### How to reproduce

1. Open the page
2. Open the offcanvas cart (modal dialog)
3. Use the Tab key to navigate forward through interactive elements inside the modal
4. Continue tabbing after reaching the last focusable element

**Comment by tobiasberge** [comment]
Hi @MichalMG, 

thanks for the report. I have also noticed this and can reproduce it.

This seems to be related to the focus-trap behaviour in Bootstrap. Our Off-Canvas dialog is created via JS and is added at the end of the document. But when there is no other focusable element after the Off-Canvas dialog, the standard focus-trap is not working. 

I have also opened an issue in the Bootstrap repository but in the meantime we will look for an alternative fix/workaround.

https://github.com/twbs/bootstrap/issues/42503
